libp2p/pubsub/gossipsub.py

```diff
@@ -75,16 +78,19 @@ def __init__(
         # Create peer --> protocol mapping
         self.peer_protocol = {}
 
-        # Create topic --> time since last publish map
-        self.time_since_last_publish = {}
-
         # Create message cache
         self.mcache = MessageCache(gossip_window, gossip_history)
 
         # Create heartbeat timer
         self.heartbeat_initial_delay = heartbeat_initial_delay
         self.heartbeat_interval = heartbeat_interval
 
+    async def run(self) -> None:
+        if self.pubsub is None:
+            raise NoPubsubAttached
+        self.manager.run_task(self.heartbeat)
```

Comment:
Should this be a `daemon` task?  Is it considered an error if the heartbeat task exited?